src/App.js

```diff
@@ -1,23 +1,90 @@
+import Sort from './components/Sort'
+import SignUp from './components/SignUp'
+import {PrivateRoute} from './util'
+import axios from 'axios'
+import { useData } from './contexts/DataContext'
+import { useAuth } from './contexts/AuthContext'
 function App() {
+  const {login , setUsers , userId} = useAuth()
+  const {dispatch} = useData()
+  //const id = JSON.parse(localStorage?.getItem('userId'))?.id
+  useEffect(()=>{
+    (async () => {
+        try {
+            dispatch({type : 'LOADING_STATUS' , payload : true});
+            const {data : {products}} = await axios.get('https://intense-scrubland-09454.herokuapp.com/products')
+            dispatch({type : 'DATA' , payload : products})
+        } catch (error) {
+            console.log(error)
+        }
+        finally {
+            dispatch({type : 'LOADING_STATUS' , payload : false});
+        }
+    })(); 
+},[dispatch])
+useEffect (() => {
+  (async () => {
+          try {
+              const {data : {user}} = await axios.get('https://intense-scrubland-09454.herokuapp.com/user')
+              setUsers(user)
+          } catch (error) {
+              
+          }
+      })()
+},[login , setUsers])
+useEffect(() => {
+  if(userId){
+  (async () => {
+      try {
+          const {data : {cart}} = await axios.get(`https://intense-scrubland-09454.herokuapp.com/cart/${userId}`)
+          console.log(cart)
+          dispatch({type : 'LOAD_CART' , payload : cart})
+      } catch (error) {
+          console.log(error)
+      }          
+  })()
+}
+},[dispatch , userId])
+useEffect(() =>{
+  if(userId){
+      (async () => {
+          try {
+              const {data : {wishlist}} = await axios.get(`https://intense-scrubland-09454.herokuapp.com/wishlist/${userId}`)
+              console.log(wishlist)
+              dispatch({type : 'LOAD_WISHLIST' , payload : wishlist})
+          } catch (error) {
+              console.log(error)
+          }          
+      })()
+}},[dispatch , userId])
   return (
     <div className="App">
-      <header className="App-header">
-        <img src={logo} className="App-logo" alt="logo" />
-        <p>
-          Edit <code>src/App.js</code> and save to reload.
-        </p>
-        <a
-          className="App-link"
-          href="https://reactjs.org"
-          target="_blank"
-          rel="noopener noreferrer"
-        >
-          Learn React
-        </a>
-      </header>
+        <Routes>
+          <Route path = '/' element = {<Home />} />
+          <Route path = '/productlist' element = {<ProductList />} />
+          <Route path = '/productlist/:productId' element = {<Productpage />} />
+          <PrivateRoute path ='/wishlist' element = {<WishList />} />
+          <PrivateRoute path = '/cart' element = {<Cart />} />
+          <Route path = '/login' element = {<Login />} />
+          <Route path = '/filters' element = {<Filter />} />
+          <Route path = '/sort' element = {<Sort />} />
```

Comment:
add filters & sort elements on the listing page !

Reply:
No need that component is fir desktop and it is there as a sidebar